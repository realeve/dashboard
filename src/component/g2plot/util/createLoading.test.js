import React from 'react';
import ChartLoading from './createLoading';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

// umi test ./src/component/g2plot/util/createLoading.test.js
// 详情可参考学习：https://enzymejs.github.io/enzyme/docs/installation/index.html
test('<ChartLoading/>', () => {
  const { container, rerender } = render(<ChartLoading type="bar" />);

  expect(container.querySelector('title')).toHaveTextContent('Loading...');

  rerender(<ChartLoading loadingTemplate={'a'} />);
  expect(container.firstChild.innerHTML).toBe('a');

  rerender(<ChartLoading type="unknown" />);
  expect(container.querySelectorAll('rect')).toHaveLength(11);
});
