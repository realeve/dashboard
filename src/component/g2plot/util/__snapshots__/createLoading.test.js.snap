// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChartLoading/> 1`] = `"<div style=\\"position: absolute; width: 100%; height: 100%; display: flex; align-items: center; justify-content: center; left: 0px; top: 0px; z-index: 99; border-radius: 4px; background-color: rgba(0, 0, 0, 0.2);\\"><svg aria-labelledby=\\"mrcdl3j-aria\\" role=\\"img\\" viewBox=\\"0 0 400 180\\" width=\\"200\\" height=\\"90\\"><title id=\\"mrcdl3j-aria\\">Loading...</title><rect role=\\"presentation\\" x=\\"0\\" y=\\"0\\" width=\\"100%\\" height=\\"100%\\" clip-path=\\"url(#mrcdl3j-diff)\\" style=\\"fill: url(#mrcdl3j-animated-diff);\\"></rect><defs role=\\"presentation\\"><clipPath id=\\"mrcdl3j-diff\\"><rect fill=\\"#dddddd\\" x=\\"20\\" y=\\"5\\" rx=\\"0\\" ry=\\"0\\" width=\\"1\\" height=\\"170\\"></rect><rect fill=\\"#dddddd\\" x=\\"20\\" y=\\"175\\" rx=\\"0\\" ry=\\"0\\" width=\\"360\\" height=\\"1\\"></rect><rect fill=\\"#dddddd\\" x=\\"40\\" y=\\"75\\" rx=\\"0\\" ry=\\"0\\" width=\\"35\\" height=\\"100\\"></rect><rect fill=\\"#dddddd\\" x=\\"80\\" y=\\"125\\" rx=\\"0\\" ry=\\"0\\" width=\\"35\\" height=\\"50\\"></rect><rect fill=\\"#dddddd\\" x=\\"120\\" y=\\"105\\" rx=\\"0\\" ry=\\"0\\" width=\\"35\\" height=\\"70\\"></rect><rect fill=\\"#dddddd\\" x=\\"160\\" y=\\"35\\" rx=\\"0\\" ry=\\"0\\" width=\\"35\\" height=\\"140\\"></rect><rect fill=\\"#dddddd\\" x=\\"200\\" y=\\"55\\" rx=\\"0\\" ry=\\"0\\" width=\\"35\\" height=\\"120\\"></rect><rect fill=\\"#dddddd\\" x=\\"240\\" y=\\"15\\" rx=\\"0\\" ry=\\"0\\" width=\\"35\\" height=\\"160\\"></rect><rect fill=\\"#dddddd\\" x=\\"280\\" y=\\"135\\" rx=\\"0\\" ry=\\"0\\" width=\\"35\\" height=\\"40\\"></rect><rect fill=\\"#dddddd\\" x=\\"320\\" y=\\"85\\" rx=\\"0\\" ry=\\"0\\" width=\\"35\\" height=\\"90\\"></rect></clipPath><linearGradient id=\\"mrcdl3j-animated-diff\\"><stop offset=\\"0%\\" stop-color=\\"#f5f6f7\\" stop-opacity=\\"1\\"><animate attributeName=\\"offset\\" values=\\"-2; -2; 1\\" keyTimes=\\"0; 0.25; 1\\" dur=\\"1s\\" repeatCount=\\"indefinite\\"></animate></stop><stop offset=\\"50%\\" stop-color=\\"#eee\\" stop-opacity=\\"1\\"><animate attributeName=\\"offset\\" values=\\"-1; -1; 2\\" keyTimes=\\"0; 0.25; 1\\" dur=\\"1s\\" repeatCount=\\"indefinite\\"></animate></stop><stop offset=\\"100%\\" stop-color=\\"#f5f6f7\\" stop-opacity=\\"1\\"><animate attributeName=\\"offset\\" values=\\"0; 0; 3\\" keyTimes=\\"0; 0.25; 1\\" dur=\\"1s\\" repeatCount=\\"indefinite\\"></animate></stop></linearGradient></defs></svg></div>"`;

exports[`<ChartLoading/> 2`] = `"<div style=\\"position: absolute; width: 100%; height: 100%; display: flex; align-items: center; justify-content: center; left: 0px; top: 0px; z-index: 99; border-radius: 4px; background-color: rgba(0, 0, 0, 0.2);\\"><svg aria-labelledby=\\"f9xpato-aria\\" role=\\"img\\" viewBox=\\"0 0 400 180\\" width=\\"200\\" height=\\"90\\"><title id=\\"f9xpato-aria\\">Loading...</title><rect role=\\"presentation\\" x=\\"0\\" y=\\"0\\" width=\\"100%\\" height=\\"100%\\" clip-path=\\"url(#f9xpato-diff)\\" style=\\"fill: url(#f9xpato-animated-diff);\\"></rect><defs role=\\"presentation\\"><clipPath id=\\"f9xpato-diff\\"><rect fill=\\"#dddddd\\" x=\\"20\\" y=\\"5\\" rx=\\"0\\" ry=\\"0\\" width=\\"1\\" height=\\"170\\"></rect><rect fill=\\"#dddddd\\" x=\\"20\\" y=\\"175\\" rx=\\"0\\" ry=\\"0\\" width=\\"360\\" height=\\"1\\"></rect><rect fill=\\"#dddddd\\" x=\\"40\\" y=\\"75\\" rx=\\"0\\" ry=\\"0\\" width=\\"35\\" height=\\"100\\"></rect><rect fill=\\"#dddddd\\" x=\\"80\\" y=\\"125\\" rx=\\"0\\" ry=\\"0\\" width=\\"35\\" height=\\"50\\"></rect><rect fill=\\"#dddddd\\" x=\\"120\\" y=\\"105\\" rx=\\"0\\" ry=\\"0\\" width=\\"35\\" height=\\"70\\"></rect><rect fill=\\"#dddddd\\" x=\\"160\\" y=\\"35\\" rx=\\"0\\" ry=\\"0\\" width=\\"35\\" height=\\"140\\"></rect><rect fill=\\"#dddddd\\" x=\\"200\\" y=\\"55\\" rx=\\"0\\" ry=\\"0\\" width=\\"35\\" height=\\"120\\"></rect><rect fill=\\"#dddddd\\" x=\\"240\\" y=\\"15\\" rx=\\"0\\" ry=\\"0\\" width=\\"35\\" height=\\"160\\"></rect><rect fill=\\"#dddddd\\" x=\\"280\\" y=\\"135\\" rx=\\"0\\" ry=\\"0\\" width=\\"35\\" height=\\"40\\"></rect><rect fill=\\"#dddddd\\" x=\\"320\\" y=\\"85\\" rx=\\"0\\" ry=\\"0\\" width=\\"35\\" height=\\"90\\"></rect></clipPath><linearGradient id=\\"f9xpato-animated-diff\\"><stop offset=\\"0%\\" stop-color=\\"#f5f6f7\\" stop-opacity=\\"1\\"><animate attributeName=\\"offset\\" values=\\"-2; -2; 1\\" keyTimes=\\"0; 0.25; 1\\" dur=\\"1s\\" repeatCount=\\"indefinite\\"></animate></stop><stop offset=\\"50%\\" stop-color=\\"#eee\\" stop-opacity=\\"1\\"><animate attributeName=\\"offset\\" values=\\"-1; -1; 2\\" keyTimes=\\"0; 0.25; 1\\" dur=\\"1s\\" repeatCount=\\"indefinite\\"></animate></stop><stop offset=\\"100%\\" stop-color=\\"#f5f6f7\\" stop-opacity=\\"1\\"><animate attributeName=\\"offset\\" values=\\"0; 0; 3\\" keyTimes=\\"0; 0.25; 1\\" dur=\\"1s\\" repeatCount=\\"indefinite\\"></animate></stop></linearGradient></defs></svg></div>"`;
